lib/src/sidenav/Sidenav.tsx

```diff
@@ -67,49 +158,163 @@ const SidenavContainer = styled.div<SidenavPropsType>`
+  font-size: ${(props) => props.theme.titleFontSize};
+  line-height: 27px;
+  display: flex;
+  align-items: center;
   color: ${(props) => props.theme.titleFontColor};
-  letter-spacing: ${(props) => props.theme.titleFontLetterSpacing};
   text-transform: ${(props) => props.theme.titleFontTextTransform};
-  margin-bottom: 16px;
+  letter-spacing: ${(props) => props.theme.titleFontLetterSpacing};
+  padding: 0.5rem 1.2rem;
+
+  svg {
+    margin-right: 0.5rem;
+  }
 `;
 
-const SidenavMenuSubTitle = styled.div`
-  font-family: ${(props) => props.theme.subtitleFontFamily};
-  font-size: ${(props) => props.theme.subtitleFontSize};
-  font-style: ${(props) => props.theme.subtitleFontStyle};
-  font-weight: ${(props) => props.theme.subtitleFontWeight};
-  color: ${(props) => props.theme.subtitleFontColor};
-  letter-spacing: ${(props) => props.theme.subtitleFontLetterSpacing};
-  text-transform: ${(props) => props.theme.subtitleFontTextTransform};
-  margin-bottom: 4px;
+const Divider = styled.div`
+  width: 100%;
+  height: 1px;
+  background-color: #999999;
 `;
 
-const SidenavMenuLink = styled.a`
-  font-family: ${(props) => props.theme.linkFontFamily};
-  font-size: ${(props) => props.theme.linkFontSize};
-  font-style: ${(props) => props.theme.linkFontStyle};
-  font-weight: ${(props) => props.theme.linkFontWeight};
-  color: ${(props) => props.theme.linkFontColor};
+const SidenavGroup = styled.div`
+  width: 100%;
+  a {
+    padding: 0.5rem 1.2rem 0.5rem 2.25rem;
+  }
+`;
+
+const SidenavGroupTitle = styled.span`
+  svg {
+    width: 16px;
+    margin-right: 0.5rem;
+  }
+  box-sizing: border-box;
+  width: 100%;
+
+  font-family: ${(props) => props.theme.groupTitleFontFamily};
+  font-style: ${(props) => props.theme.groupTitleFontStyle};
+  font-weight: ${(props) => props.theme.groupTitleFontWeight};
+  font-size: ${(props) => props.theme.groupTitleFontSize};
+  line-height: 18px;
+
+  display: flex;
+  align-items: center;
+  margin: 0px;
+  padding: 0.5rem 1.2rem;
+`;
+
+const SidenavGroupTitleButton = styled.button`
+  all: unset;
+  cursor: pointer;
+  justify-content: space-between;
+  box-sizing: border-box;
+  width: 100%;
+
+  font-family: ${(props) => props.theme.groupTitleFontFamily};
+  font-style: ${(props) => props.theme.groupTitleFontStyle};
+  font-weight: ${(props) => props.theme.groupTitleFontWeight};
+  font-size: ${(props) => props.theme.groupTitleFontSize};
+  line-height: 19px;
+
+  display: flex;
+  align-items: center;
+  margin: 0px;
+  padding: 0.5rem 1.2rem;
+  &:focus {
+    outline: 2px solid ${(props) => props.theme.linkFocusColor};
+    outline-offset: -2px;
+  }
+  &:hover {
+    background-color: ${(props) => props.theme.groupTitleHoverBackgroundColor};
+  }
+  &:active {
+    background-color: ${(props) => props.theme.groupTitleActiveBackgroundColor};
+  }
+
+  svg {
+    width: 18px;
+    height: auto;
+  }
+`;
+
+type StyledLinkProps = {
+  selected: boolean;
+};
```

Comment:
I think it is better to remove this type and indicate it directly in the styled component:
`styled.a<{ selected: boolean; }>`